### PR TITLE
Add app-emacs/ws-butler and app-emacs/kaolin-themes

### PR DESCRIPTION
**Pull request by qookei**

### CODE REVIEW

File: app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit elisp
+
+DESCRIPTION="Set of eye pleasing themes for GNU Emacs"
+HOMEPAGE="https://github.com/ogdenwebb/emacs-kaolin-themes"
+SRC_URI="https://github.com/ogdenwebb/emacs-kaolin-themes/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (xgqt) [inline app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild:10]: live 9999 support would be nice

File: app-emacs/kaolin-themes/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
```
Comment (xgqt) [inline app-emacs/kaolin-themes/metadata.xml:3]: add gnu-emacs@gentoo.org plz

File: app-emacs/ws-butler/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
```
Comment (xgqt) [inline app-emacs/ws-butler/metadata.xml:3]: maint, as above

File: app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+
+SITEFILE="50${PN}-gentoo.el"
+
+src_compile() {
+	elisp-compile themes/*.el
+	elisp_src_compile
```
Comment (ulm) [inline app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild:24]: Make it all explicit please, i.e. no `elisp_src_compile`:
```suggestion
	elisp-compile *.el themes/*.el
```

File: app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+src_install() {
+	elisp-install ${PN} themes/*.el themes/*.elc
+	elisp_src_install
```
Comment (ulm) [inline app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild:29]: Same here.

File: app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit elisp
+
+DESCRIPTION="Set of eye pleasing themes for GNU Emacs"
+HOMEPAGE="https://github.com/ogdenwebb/emacs-kaolin-themes"
+SRC_URI="https://github.com/ogdenwebb/emacs-kaolin-themes/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/emacs-kaolin-themes-${PV}"
```
Comment (ulm) [inline app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild:11]: Might be worth defining `MY_PN="emacs-${PN}"` and use the variable for `SRC_URI` and `S`.

File: app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+DEPEND="app-emacs/autothemer"
+RDEPEND="${DEPEND}"
```
Comment (ulm) [inline app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild:14]: The dependency is needed on the build host.
```suggestion
RDEPEND="app-emacs/autothemer"
BDEPEND="${RDEPEND}"
```

File: app-emacs/ws-butler/ws-butler-0.6_p20201117.ebuild [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+EAPI=8
+
+inherit elisp
+
+COMMIT_HASH="e3a38d93e01014cd47bf5af4924459bd145fd7c4"
+
+DESCRIPTION="Unobtrusively trim extraneous white-space *ONLY* in lines edited"
+HOMEPAGE="https://github.com/lewang/ws-butler"
+SRC_URI="https://github.com/lewang/ws-butler/archive/${COMMIT_HASH}.tar.gz -> ${P}.tar.gz"
```
Comment (ulm) [inline app-emacs/ws-butler/ws-butler-0.6_p20201117.ebuild:12]:
```suggestion
SRC_URI="https://github.com/lewang/${PN}/archive/${COMMIT_HASH}.tar.gz -> ${P}.tar.gz"
```

**Review by ulm — CHANGES_REQUESTED** [review]

**Comment by qookei** [comment]
I've resolved the comments.

**Comment by xgqt** [comment]
ugh, I forgot to "sumbit" my review again, sorry

File: app-emacs/ws-butler/metadata.xml [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  <maintainer type="person" proxied="yes">
+    <email>kacper.slominski72@gmail.com</email>
+    <name>Kacper Słomiński</name>
+  </maintainer>
+  <maintainer type="project" proxied="proxy">
+    <email>proxy-maint@gentoo.org</email>
+    <name>Proxy Maintainers</name>
+  </maintainer>
+  <upstream>
```
[the remark this answers is not in the document]
Comment (xgqt) [reply app-emacs/ws-butler/metadata.xml:16]: > add `<bugs-to>` based on remote-id

https://github.com/lewang/ws-butler/issues/

File: app-emacs/ws-butler/ws-butler-0.6_p20201117.ebuild [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+	inherit git-r3
+	EGIT_REPO_URL="https://github.com/lewang/${PN}.git"
+else
+	SRC_URI="https://github.com/lewang/${PN}/archive/${COMMIT_HASH}.tar.gz -> ${P}.tar.gz"
+	S="${WORKDIR}/${PN}-${COMMIT_HASH}"
+fi
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
```
Comment (xgqt) [inline app-emacs/ws-butler/ws-butler-0.6_p20201117.ebuild:23]:
```suggestion
KEYWORDS="~amd64 ~x86"
```

File: app-emacs/ws-butler/metadata.xml [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+  <maintainer type="project" proxied="proxy">
+    <email>proxy-maint@gentoo.org</email>
+    <name>Proxy Maintainers</name>
+  </maintainer>
+  <maintainer type="project">
+    <email>gnu-emacs@gentoo.org</email>
+    <name>Gentoo GNU Emacs project</name>
+  </maintainer>
+  <upstream>
+    <remote-id type="github">lewang/ws-butler</remote-id>
```
Comment (xgqt) [inline app-emacs/ws-butler/metadata.xml:17]:
```suggestion
    <remote-id type="github">lewang/ws-butler</remote-id>
    <bugs-to>https://github.com/lewang/ws-butler/issues/</bugs-to>
```

File: app-emacs/kaolin-themes/metadata.xml [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+  <maintainer type="project" proxied="proxy">
+    <email>proxy-maint@gentoo.org</email>
+    <name>Proxy Maintainers</name>
+  </maintainer>
+  <maintainer type="project">
+    <email>gnu-emacs@gentoo.org</email>
+    <name>Gentoo GNU Emacs project</name>
+  </maintainer>
+  <upstream>
+    <remote-id type="github">ogdenwebb/emacs-kaolin-themes</remote-id>
```
Comment (xgqt) [inline app-emacs/kaolin-themes/metadata.xml:17]:
```suggestion
    <bugs-to>https://github.com/ogdenwebb/emacs-kaolin-themes/issues/</bugs-to>
    <remote-id type="github">ogdenwebb/emacs-kaolin-themes</remote-id>
```

File: app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+EAPI=8
+
+inherit elisp
+
+DESCRIPTION="Set of eye pleasing themes for GNU Emacs"
+HOMEPAGE="https://github.com/ogdenwebb/emacs-kaolin-themes"
+
+MY_PN="emacs-${PN}"
+
+if [[ "${PV}" == "9999" ]]; then
```
Comment (xgqt) [inline app-emacs/kaolin-themes/kaolin-themes-1.6.9.ebuild:13]: `== *9999*` is more widely used but this is fine too :)

**Comment by qookei** [comment]
Resolved the comments again :^).

**Comment by xgqt** [comment]
> Resolved the comments again :^).

Thanks a lot, I will merge it in a second.

**Pull request closed without merging**